### DISCUSSION TITLE
[Python] Use alpine on for containers : was it a good idea ?

### DISCUSSION THREAD

**Discussion by waghanza** [the-benchmarker/web-frameworks#4321] [Show and tell]
Hi @ahopkins,

Some wheels make troubles on `alpine`. Is this a good idea, in **production** to use `alpine` when using containers.

I've also found https://pythonspeed.com/articles/alpine-docker-python/

What do you think @the-benchmarker/web-frameworks ?

Regards,

**Comment by ahopkins** [comment]
I would say that the Python community seems to be moving away from Alpine. While I have used it in production in the past to run python apps, it seems to be falling out of favor. I say this having done no actual research on the topic, and take this from the small sample size of the people I interact with, and the general impression I have seen. Debian slim seems to be favored.

**Comment by dalisoft** [comment]
I used all variants on production and most performed well is `ubuntu`/`debian` full docker image. On slim/alpine version some functions like `ffmpeg`,`tensorflow` does not work properly or i am not able to-do properly

**Comment by waghanza** [comment]
The official documentation point slim variant https://docs.docker.com/language/python/build-images/

**Comment by RobertoPrevato** [comment]
I like Alpine, but I had often abandoned it due to compatibility issues, as already mentioned above. For example somebody asked why I used Ubuntu for my [custom DevOps agents](https://github.com/RobertoPrevato/AzureDevOps-agents) - good luck installing all build tools on Alpine (official agents from MS use Ubuntu and also GitHub Actions use the same)! I use `slim` when I need to deploy a Python application in Docker.
